events/controller/task.go

```diff
@@ -280,3 +283,96 @@ func (n *NatsConditionTaskRepository) Query(ctx context.Context) (*condition.Tas
 
 	return task, nil
 }
+
+// HTTPTaskRepository implements the ConditionTaskRepository to Publish and Query Task information from NATS over HTTP.
+type HTTPTaskRepository struct {
+	logger        *logrus.Logger
+	orcQueryor    orc.Queryor
+	conditionKind condition.Kind
+	conditionID   uuid.UUID
+	serverID      uuid.UUID
+}
+
+func NewHTTPTaskRepository(
+	serverID,
+	conditionID uuid.UUID,
+	conditionKind condition.Kind,
+	orcQueryor orc.Queryor,
+	logger *logrus.Logger,
+) ConditionTaskRepository {
+	return &HTTPTaskRepository{
+		orcQueryor:    orcQueryor,
+		conditionID:   conditionID,
+		conditionKind: conditionKind,
```

Comment:
why embed the kind on the publisher? A publisher could handle many kinds.

Reply:
The controller initializes a `Publisher` for each Condition it receives, the `Publisher` is kept around only in the context of the Condition as long as its being processed. This is intentional, especially for how the NATS Publisher is bound to a specific KV key, I don't see any reason to reuse Publishers among Conditions.